### PR TITLE
Pass workspace context to external MLflow UI on launch

### PR DESCRIPTION
**Pull request by nananosirova**
https://redhat.atlassian.net/browse/RHOAIENG-60053

## Description

Launching the external MLflow UI via the "Launch MLflow" button caused the user to lose their current project context. The button always opened a fixed `/mlflow` URL without any workspace information, so the external MLflow UI had no way to know which project the user was working in.
This PR passes the active workspace as a hash fragment query parameter so the external MLflow UI lands in the correct project.

## How Has This Been Tested?

Manually verified on cluster that clicking "Launch MLflow" from the Experiments and Prompts pages preserves workspace context.

## Test Impact

Added unit tests.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **New Features**
  * MLflow launch button now generates workspace-specific URLs to provide proper context when launching MLflow from different workspaces.

* **Tests**
  * Added comprehensive tests covering workspace route generation, URL encoding, and edge cases for MLflow launch functionality.

### CODE REVIEW

**Comment by srtanish1992** [comment]
/lgtm

Code changes LGTM. Manually tested and work as expected.

The only difference I see is, in dashboard we show project display name in project selector and in MLflow we show the kubernetes namespace/workspace so they do not match.

https://github.com/user-attachments/assets/430ff61c-8cd7-4b46-9473-e94001c17363

File: frontend/src/pages/projects/screens/detail/overview/trainModels/__tests__/MLflowCard.spec.ts [hunk 1]
```diff
@@ -43,6 +44,30 @@ describe('setWorkspaceQueryParam', () => {
   });
 });
 
+describe('mlflowLaunchRoute', () => {
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/projects/screens/detail/overview/trainModels/__tests__/MLflowCard.spec.ts:47]: These tests are for `mlflowLaunchRoute`, which lives in `frontend/src/routes/pipelines/mlflow.ts`. Per the project's co-location convention, they'd be better placed in `frontend/src/routes/pipelines/__tests__/mlflow.spec.ts`.

Not blocking since the tests themselves are solid, but worth moving in this PR or a quick follow-up so people looking for route tests find them in the expected place.

nit: CodeRabbit flagged this too -- adding a `mlflowLaunchRoute('')` case would be a nice addition since both callers default `workspace` to `''` via `searchParams.get(...) ?? ''`. The `!''` check in JS is truthy (so it'd return `/mlflow`), which is the right behavior, but documenting that in a test makes the intent explicit.

File: frontend/src/routes/pipelines/mlflow.ts [hunk 2]
```diff
@@ -32,3 +32,10 @@ export const globPromptManagementAll = `${promptManagementPath}/*`;
 
 export const mlflowPromptManagementBaseRoute = (namespace?: string): string =>
   withWorkspace(promptManagementPath, namespace);
+
+export const mlflowLaunchRoute = (namespace?: string): string => {
+  if (!namespace) {
+    return MLFLOW_PROXY_BASE_PATH;
+  }
+  return `${MLFLOW_PROXY_BASE_PATH}/#/?${WORKSPACE_QUERY_PARAM}=${encodeURIComponent(namespace)}`;
```
Comment (DaoDaoNoCode) [inline frontend/src/routes/pipelines/mlflow.ts:40]: The URL structure here -- `/mlflow/#/?workspace=...` -- differs from `withWorkspace` above (which uses `?workspace=...` without the hash fragment). That's intentional because the external MLflow UI uses hash-based routing while the embedded pages use regular routing, so this makes sense.

Just wanted to confirm: does the external MLflow UI actually read `workspace` from `window.location.hash` at the root path (`/#/?workspace=...`)? `buildMLflowExperimentsWorkspaceHref` in MLflowCard.tsx builds `/#/experiments?workspace=...` instead (with a sub-path in the hash). If the external UI only looks for workspace on specific hash routes like `/#/experiments`, this root-level hash might not be picked up.

nit: `encodeURIComponent` is overkill for Kubernetes namespace names (RFC 1123: lowercase alphanumeric + hyphens only), but it's the right defensive choice -- no harm in encoding.

**Review by DaoDaoNoCode — COMMENTED** [review]
Clean, focused change that threads the workspace through the LaunchMlflowButton so the external MLflow UI opens with the right project context. Logic is straightforward, tests cover the new function well, all CI checks pass.

Local checks: lint (0 errors) -- type-check -- unit tests -- all green.

Left a couple of comments inline -- one about test file placement and one question about the hash-based URL structure. Also one broader observation below about a related link in MLflowCard.tsx (not part of this diff) that could be a quick follow-up.

**Comment by DaoDaoNoCode** [comment]
~Side observation (not part of this diff): In `MLflowCard.tsx` line 81, the "Go to Experiments" link navigates to `mlflowExperimentsPath` (`/develop-train/mlflow/experiments`) without passing the current project's workspace. Since the user is on a project details page at that point, they'd lose their project context when clicking that link. Using `mlflowExperimentsBaseRoute(projectName)` there would carry the workspace forward. Totally fine as a follow-up since it's pre-existing behavior.~
Invalid AI review. Removed

**Review by DaoDaoNoCode — APPROVED** [review]
Clean fix, LGTM
/lgtm
